### PR TITLE
Update the game for some recent changes to the framework

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: lib/render_coordinate_system.dart [hunk 1]
```diff
@@ -85,7 +85,7 @@ class RenderCoordinateSystem extends RenderProxyBox {
     }
   }
 
-  void applyPaintTransform(Matrix4 transform) {
+  void applyPaintTransform(RenderObject child, Matrix4 transform) {
```
Comment (abarth) [inline lib/render_coordinate_system.dart]: @Hixie Is this correct?

File: lib/render_coordinate_system.dart [hunk 2]
```diff
@@ -85,7 +85,7 @@ class RenderCoordinateSystem extends RenderProxyBox {
...
     super.applyPaintTransform(transform);
     transform.multiply(_effectiveTransform);
```
Comment (Hixie) [inline lib/render_coordinate_system.dart]: iirc these two lines need reversing

File: lib/render_coordinate_system.dart [hunk 3]
```diff
@@ -85,7 +85,7 @@ class RenderCoordinateSystem extends RenderProxyBox {
     }
   }
 
-  void applyPaintTransform(Matrix4 transform) {
+  void applyPaintTransform(RenderObject child, Matrix4 transform) {
     super.applyPaintTransform(transform);
```
Comment (Hixie) [inline lib/render_coordinate_system.dart]: don't forget to update this call too

**Comment by Hixie** [comment]
we really need to add flutter sprites to the Travis build

File: lib/render_coordinate_system.dart (hunk 3 above)
Comment (jason-simmons) [inline lib/render_coordinate_system.dart]: I've never seen this method be invoked, but the declaration needed to be fixed in order to pass type checking.

@vlidholt: is this function still necessary?

Comment (vlidholt) [inline lib/render_coordinate_system.dart]: I don't think this is needed anymore

**Comment by vlidholt** [comment]
Yes, I'll see if I can set up some tests next week, I think that would be are really good idea!

File: lib/render_coordinate_system.dart (hunk 3 above)
Comment (jason-simmons) [inline lib/render_coordinate_system.dart]: Removed

**Comment by vlidholt** [comment]
As far as I can tell this LGTM, but perhaps someone else should also have a look.

File: lib/render_coordinate_system.dart [hunk 4]
```diff
@@ -85,11 +85,6 @@ class RenderCoordinateSystem extends RenderProxyBox {
     }
   }
 
-  void applyPaintTransform(Matrix4 transform) {
```
Comment (Hixie) [inline lib/render_coordinate_system.dart]: This is definitely needed, it's part of the RenderObject protocol.

This whole file should move to the rendering library though.

Comment (jason-simmons) [inline lib/render_coordinate_system.dart]: Brought this function back and changed it to do the transform.multiply() before calling the superclass

**Pull request merged**